serialzy/api.py

```diff
@@ -39,8 +39,9 @@ class VersionBoundary:
 class Serializer(abc.ABC):
     HEADER_BYTES = 'serialzy'.encode('utf-8')
     HEADER_BYTES_LEN = len(HEADER_BYTES)
+    _USER_META_KEY = 'user-meta'
 
-    def serialize(self, obj: Any, dest: BinaryIO) -> None:
+    def serialize(self, obj: Any, dest: BinaryIO, user_meta: Optional[Dict[str, Any]] = None) -> None:
```

Comment:
I'm not sure but maybe it is a good idea to make an alias for `UserMeta = Dict[str, Any]` to simplify the code and make it more readable.